### ISSUE TITLE
Typing of universally quantified record fields

### ISSUE THREAD

**Issue by xclerc** [ocaml/ocaml#8556]
Given the following interface:
```
val length : 'a list -> int
```

An implementation going through a record field does
not compile with 4.08 (it compiles with 4.07):
```
type t = { length : 'a . 'a list -> int; }
let get_length () = { length = List.length; }
let { length; } = get_length ()
```

```
Error: The implementation c.ml does not match the interface c.cmi:
       Values do not match:
         val length : '_weak1 list -> int
       is not included in
         val length : 'a list -> int
```

A trivial workaround is to go through a "temporary" variable: 
```
type t = { length : 'a . 'a list -> int; }
let get_length () = { length = List.length; }
let tmp = get_length ()
let length = tmp.length
```

**Comment by gasche** [comment]
This sounds very much like the issue that was fixed by @trefis in #2317, merged 4 days ago. Are your tests more recent than that?

**Comment by xclerc** [comment]
Indeed, sorry for the noise, I mixed it up with the patch for https://github.com/ocaml/ocaml/issues/8548.
